frontend/packages/console-app/src/components/cloud-shell/CloudShellTab.tsx

```diff
@@ -1,11 +1,17 @@
 import * as React from 'react';
 import { useTranslation } from 'react-i18next';
-import { InlineTechPreviewBadge } from '@console/shared';
+import { Redirect } from 'react-router';
+import { InlineTechPreviewBadge, useFlag } from '@console/shared';
+import { FLAG_DEVWORKSPACE } from '../../consts';
 import CloudShellTerminal from './CloudShellTerminal';
 import './CloudShellTab.scss';
 
 const CloudShellTab: React.FC = () => {
   const { t } = useTranslation();
+  const devWorkspaceFlag = useFlag(FLAG_DEVWORKSPACE);
+
+  if (!devWorkspaceFlag) return <Redirect to="/" />;
```

Comment:
is it better to handle this at the Route level instead of the component 
Changing it in the plugin

Reply:
How do you suggest we do this at the route level? This route is in the root level `<App>` component which also renders the main redux store context which contains all the flag information. Do you have a solution for this in mind?

Reply:
I missed that it was placed beside the '/' at the route level. It is still doable with a manual API fetch for checkng the presence of the CRD using the model, but that would again slow it down I guess
Well, then the `<Redirect/>` is the best approach in this case.